hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -557,14 +558,23 @@ def get_element(self, i):
         return self._transform(self.elements[i])
 
     def do_filtered_draw(self, data):
+        # Bundle classes shrink towards the end of the bundle to focus on the more recently generated examples
+        shrink_target = (
+            len(self.elements)
+            if self.__class__.__name__ in ["Bundle", "BundleConsumer"]
+            else 0
+        )
```

Comment:
Instead of conditioning on the type of self here, let's define a classvar `shrink_towards = 0` and then we can define `class Bundle(SampledFromStrategy[Ex]):` with `shrink_towards = float("inf")`  (we'll also need to adjust the type annotations on `HypothesisProvider.draw_integer()` but that's OK)

Alternatively, if the current approach to shrinking-from-the-back turns out to be too tricky, we could try making a `Bundle` _have a_ `SampledFromStrategy`, rather than _being a_.  Off the top of my head we'd lose some efficiency stuff for unique lists, but it's dubious whether those are valid for bundles anyway (since elements can be both added and consumed).